pkg/vault/contracts/SingletonAuthentication.sol

```diff
@@ -16,6 +16,11 @@ import { Authentication } from "@balancer-labs/v3-solidity-utils/contracts/helpe
 abstract contract SingletonAuthentication is Authentication {
     IVault private immutable _vault;
 
+    modifier onlySwapFeeManagerOrAuthentication(address pool) {
```

Comment:
Maybe `onlySwapFeeManagerOrAuthenticate` / `onlySwapFeeManagerXorAuthenticate` sounds a bit better (xor is what we are actually doing, although the name doesn't sound great. Leaving that to @EndymionJkb :) )

Reply:
If we don't want to use exclusiveRole terminology, and have to make it swap-manager specific, maybe `onlySwapFeeManagerOrGovernance`, or `onlySwapFeeManagerIfDefined`?

I'd still prefer to have the `_ensureAuthenticatedByExclusiveRole` function parameterized by the role, and then give the modifiers more specific names, and have them call the generic role authentication (identical to VaultAdmin). It would be most consistent across the whole codebase. Otherwise we're making up different patterns and naming conventions when there's already a suitable one there.